### DISCUSSION TITLE
Issue with useAccount not updating after connecting Web3Modal with Wagmi

### DISCUSSION THREAD

**Discussion by victorpavlenko** [reown-com/appkit#2681] [Q&A]
I upgraded from **v2** to **v5** and encountered an issue when using `@web3modal/wagmi/react`. 

After invoking the modal for wallet connection, `useAccount` does not update, even after the wallet is successfully connected from modal and confirmed. The state only updates after refreshing the page and waiting for some time.

I tried several approaches to resolve the issue:

Used `defaultWagmiConfig` from the `@web3modal/wagmi/react/config` package.

Tried using `createConfig` from the latest version of `wagmi@2.12.4` , and manually invoking `connect` with the **walletConnect** connector.

Also data on localhost updates, but **wagmi** does not respond to the state changes.

```
@w3m/connected_connector: WALLET_CONNECT
@w3m/recent: [{"id":"c03dfee351b6fcc421b4494ea33b9d4b92a984f87aa76d1663bb28705e95011a","name":"Uniswap Wallet","homepage":"https://uniswap.org","image_id":"bff9cf1f-df19-42ce-f62a-87f04df13c00","order":60,"mobile_link":"uniswap://","desktop_link":null,"link_mode":"https://uniswap.org/app","webapp_link":null,"app_store":"https://apps.apple.com/us/app/uniswap-wallet/id6443944476","play_store":"https://play.google.com/store/apps/details?id=com.uniswap.mobile","rdns":null,"chrome_store":null,"injected":null,"chains":["eip155:1","eip155:10","eip155:137","eip155:42161"],"installed":false}]
```

Maybe related from console
`Element w3m-connecting-wc-qrcode scheduled an update (generally because a property was set) after an update completed, causing a new update to be scheduled. This is inefficient and should be avoided unless the next update can only be scheduled as a side effect of the previous update.`

**Comment by rtomas** [comment]
Please, share with us a minimal reproducible example so we can help you with you problem.
